### DISCUSSION TITLE
Error uploading photos to github profile.

### DISCUSSION THREAD

**Discussion by joneslee0918** [community/community#48077] [Other Feature Feedback, Questions, & Ideas] [answered]
### Select Topic Area

Product Feedback

### Body

I want to upload photo to github profile, but it has error like this.
"Something went really wrong and we can’t process that picture."
I have already changed image size and extension(jpg, png) but result is the same.
please , help me

**Comment by CodeMaster7000** [comment]
The most likely issue is that the file size is simply too big. Consider using a free image optimization tool such as [Squoosh](https://squoosh.app/) in order to compress the same image into a smaller file size.

**Reply by joneslee0918** [reply]
I did.
There was a problem with the file size. My avatar size is over 10M.
Thanks.

**Reply by sircorneilous** [reply]
my file is 1 KB and a copy of it thats scaled larger is only 4 KB and the image was scaled to 500x500 px and it won't upload it and won't work

**Reply by sircorneilous** [reply]
the copy was scaled to 500x500 px and the original is 32x32 px

**Comment by OwaisAzad** [comment, marked as answer]
---

## 1. Check the File Size  
GitHub has a 10 MB file size limit for individual files. If your image exceeds this:

- Compress or resize the image before uploading.
- Tools like TinyPNG can help reduce the size without significant quality loss.

## 2. Try a Different Browser  
Some browsers might not handle file uploads well. Try:

- Switching to **Google Chrome**, **Firefox**, or another updated browser.
- Using **incognito/private mode** can also help in some cases.

## 3. Upload to a Different Repository  
If uploading directly to your profile fails:

- Try uploading to a new or existing repository.
- Use the **raw URL** of the uploaded image to link it in your profile README.

## 4. Try a VPN (Credit: @fabiogbe, @NiktoBlox-Guest)  
Some users have resolved the issue by:

- Turning ON a VPN
- Then uploading the image again.

> “If you have an anti-virus on, try connecting to a VPN before uploading the picture.” – @fabiogbe  
> “Turn on VPN and it will be solved.” – @NiktoBlox-Guest

## 5. Rename the File (Credit: @Syed-Abdullah-G)  
Avoid file names with:

- Spaces
- Colons (:)
- Special characters
- Long timestamps (e.g., `Screenshot 2025-04-10 at 11.04.19 PM.jpg`)

**Rename the file** to something simple like `file_1.jpg` before uploading.

> “I changed the file name to file_1 and it worked.” – @Syed-Abdullah-G  
> “Thank you @Syed-Abdullah-G. This worked for me.” – @cephas001

## Thanks to the Contributors:  
- @fabiogbe  
- @NiktoBlox-Guest  
- @Syed-Abdullah-G  
- @cephas001  
- @ChicoThorn

---

**Reply by joneslee0918** [reply]
I did.
There was a problem with the file size. My avatar size is over 10M.
Thanks for your efforts.

**Reply by SheikhElMoctarG** [reply]
My avatar is less than 1MB, but i still see "Something went really wrong and we can’t process that picture."

I tried alot images but no solution

**Reply by devope** [reply]
same here

**Reply by KovacevicAleksa** [reply]
Same

**Reply by sircorneilous** [reply]
same. i scaled the same image a bunch of times and used a different browser doesn't work

**Reply by sdkintern** [reply]
i wonder if there is a current mini-outage specific to this functionality

**Reply by KovacevicAleksa** [reply]
Very possibly, a lot of people today cannot upload.

**Reply by RafaPatino01** [reply]
Same problem right now

**Reply by sircorneilous** [reply]
bruh

**Reply by sircorneilous** [reply]
@SheikhElMoctarG
same problem im having. i hope this gets fixed

**Reply by salahddik** [reply]
same problem

**Reply by SheikhElMoctarG** [reply]
That means it's general problem

**Reply by KovacevicAleksa** [reply]
Now work fine

**Reply by SheikhElMoctarG** [reply]
It's back to work again, you can change your profile picture now.

**Reply by alsmanifesto** [reply]
Not working again

**Comment by malaidev** [comment]
I have uploaded 21.1KB sized profile image, and tried in different browser, but couldn't have solve above problem.

**Reply by RamanaRaj7** [reply]
same here

**Reply by nithinK-142** [reply]
The profile picture didn't change in my case may be because of some extensions i have in my browser (chrome) .
So try opening a guest window login to github and chage your avatar there. Use github recommended extension like .jpg or .png. hope this helps.

P.S. imagesize < 10mb.

**Comment by fabiogbe** [comment]
If your antivirus software is running, you can try turning a VPN on. That could do the trick.

**Comment by sydneycamille0896** [comment]
I have tried decreasing the file size, turning on VPN, using a different browser, and moving the file onto my local harddrive instead of trying to pull from OneDrive. None have worked. Anything else I could try?

**Comment by ChicoThorn** [comment]
Sept. 1, 2023 — Cannot upload new profile picture. It is under 1MB (743KB), a .jpg image (tested and works elsewhere), on my local hard drive. I've tried all the workarounds suggested above but nothing works. Here's what happens when I try to upload a picture: If the picture is too large (over 1MB) I get the warning message about the picture being too large... so I know that part of the GitHub interface is working. Next I try to upload my new profile pic (743KB), the screen jumps for just an instant but the generic profile avatar still shows and nothing at all happens. No error message, no nothing. Could it be that GitHub is having some sort of issue? If not, can someone please help?

**Reply by fabiogbe** [reply]
if you have an anti-virus on, try connecting to a VPN before uploading the picture...

**Reply by ChicoThorn** [reply]
Yep! That did the trick! Thanks! 😊

**Reply by ChicoThorn** [reply]
Hi again -- I now regret that I installed and used a VPN to do something as trivial as changing a profile picture... here's why: I'm a Windows Insider in the Canary Channel. Every month or so Microsoft releases an ISO of the current build for download through a secure Insiders download page. After I installed the VPN to change my GitHub profile picture I became LOCKED OUT of downloading ISOs. The error message said that my account was blocked from downloading because of the VPN which they called 'suspicious' activity. I've since tried everything to clear my account, including uninstalling the VPN, but no luck. I finally resorted to a clean install of my entire computer system; but even that failed to clear my account for ISO downloading. The block is linked to your email account, not your IP Address, or other identifiers. My only workaround was to use a different Microsoft account for my ISO downloads. It irks me that my main email is now in Microsoft purgatory because of this GitHub profile debacle. GitHub needs to fix this so all these hoop-jumping barriers are removed so we can easily do what should be a very simple thing: changing our profile picture. Bottom line: Maybe a VPN isn't such a good idea after all... 😕

**Reply by fabiogbe** [reply]
Hey Chico, really sorry you experienced all that. Indeed the idea is just to use the VPN for this one-off task and then turn it off.

**Comment by NiktoBlox-Guest** [comment]
Turn on VPN and it will be solved

**Comment by SaifurRR** [comment]
file format?

**Comment by Ankitkushwaha90** [comment]
i have routing file. so does not uploading . because error show.

**Comment by ChicoThorn** [comment]
________________________________

**Comment by verhovsky** [comment]
> Please upload a picture smaller than 1 MB.

Please write the software to downsample images instead of making me google ImageMagick commands. We all have modern megapixel phones, no one is taking sub-1 MB photos anymore.

**Comment by yermulnik** [comment]
```shell
> du -sh builder.png
16K     builder.png

> file builder.png
builder.png: PNG image data, 127 x 128, 8-bit/color RGBA, non-interlaced
```
![image](https://github.com/user-attachments/assets/27b5b481-755a-4478-b0f4-4d5f99848c5d)

GitHub EMU. An infinite spinner. Both Chrome and FF. No anti-viruses and other bloatware. Can upload this same image as attachment in PR comment and similar.

**Comment by Syed-Abdullah-G** [comment]
Solution:

Don't upload your files which have names like "Screenshot 2025-04-10 at 11.04.19 PM.jpg"

I then changed the file name to "file_1" and then it worked.

**Reply by cephas001** [reply]
Thank you @Syed-Abdullah-G. This worked for me.

**Comment by woeterman94** [comment]
Error I'm getting is. Refused to get unsafe header "X-Digest-Sha256-Hmac"

**Comment by vectorlex** [comment]
Use ffmpeg -i photo.png photo1.png and upload the photo1.png
it worked for me

**Comment by amyodov** [comment]
Trying to put the social preview picture for my project. NOTHING of the above works - converting with Adobe Photoshop, ImageMagick, even ffmpeg. The picture is strictly 1280x640 as recommended. The png version is 922k (ffmpeg-version is 950k), jpg version is 350k. Tried in Chrome, Chrome anonymous mode, Safari. Even tried VPN.
But honestly, if attachment-upload.ts of Github errors with "Refused to get unsafe header X-Digest-Sha256" and then "Refused to get unsafe header X-Digest-Sha256-Hmac" - probably, the code should NOT get the unsafe headers X-Digest-Sha256 and X-Digest-Sha256-Hmac.
